lib/fluent/diagtool/diagutils.rb

```diff
@@ -363,7 +363,13 @@ def diaglogger_error(str)
     end
 
     def fluent_package?
-      File.exist?("/etc/fluent/fluentd.conf") || File.exist?("/opt/fluent/bin/fluentd")
+      if Diagtool.windows?
+        # FLUENT_PACKAGE_TOPDIR indicates installed directory, which is defined by fluent-package
+        File.exist?(File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "etc/fluent/fluentd.conf")) ||
+          File.exist?(File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "bin/fluentd"))
+      else
+        File.exist?("/etc/fluent/fluentd.conf") || File.exist?("/opt/fluent/bin/fluentd")
+      end
     end
```

Comment:
This class should not have platform-specific logic.
How about fixing `lib/fluent/diagtool/windows/diagutils.rb`.

I believe that just adding the following function to the file is enough.

```rb
      def fluent_package?
        ENV.key?("FLUENT_PACKAGE_TOPDIR")
      end
```

Reply:
Thanks. It's appropriate fix.